### ISSUE TITLE
Audio ACE Step 1.5 quality degradation in v0.12.3 (glitchy artifacts) compared to v0.12.2

### ISSUE THREAD

**Issue by MssdTheMark** [Comfy-Org/ComfyUI#12375]
### Custom Node Testing

- [x] I have tried disabling custom nodes and the issue persists (see [how to disable custom nodes](https://docs.comfy.org/troubleshooting/custom-node-issues#step-1%3A-test-with-all-custom-nodes-disabled) if you need help)

### Summary

I am experiencing a significant regression in audio generation quality when using the audio_ace_step_1_5_checkpoint template after updating from ComfyUI v0.12.2 to v0.12.3 (ComfyUI Desktop v0.8.3).

The generated tracks, particularly in the Ambient/Minimal genre, now contain glitchy, overly loud, and inappropriate sounds, or strange looping patterns. This issue is less noticeable in denser genres but completely breaks the atmosphere of sparse/minimal tracks.

### Steps to Reproduce

1. Use the audio_ace_step_1_5_checkpoint workflow.
2. Use the following generation parameters:
    - Prompt: `Electronic Minimalism: A sparse and evocative electronic track based on repetition and subtle variations in sound design. Long sustained notes, delicate textures and a sense of spaciousness are key elements.`
    - Lyrics: `[Intro – Sustained Synth Note & Minimal Percussion] [Verse 1 - Subtle Texture Variations & Evolving Pad] [Bridge - Slow Filter Sweep & Atmospheric Resonance] [Outro – Gradual Silence with Lingering Harmonics]`

    - Seed: 31 (Note: Issue is reproducible on different seeds, not specific to this one)
    - Duration: 120
    - BPM: 70
    - Time Signature: 3
    - Language: en
    - Key Scale: b minor
    - Other settings: Default internal settings (cfg_scale: 2, temperature: 0.85, top_p: 0.90, top_k: 0) were used.

### Expected Behavior
On ComfyUI v0.12.2, these parameters generate a high-quality, clean ambient track with smooth textures and appropriate atmospheric resonance.

### Actual Behavior
On ComfyUI v0.12.3, the same parameters result in a track with distinct audio artifacts:

1. Glitchy and harsh sounds that do not fit the genre.
2. Inappropriately loud bursts of noise.
3. Strange looping patterns

### Potential Cause
_(Analysis assisted by GLM 4.7)_

The regression appears to stem from changes in how the generation metadata is formatted in the ACE15 tokenizer (`.\ComfyUI\comfy\text_encoders\ace15.py`).

1. Redundant Context Injection: The user_metas dictionary incorrectly includes "caption". This appends the user's text prompt to the Assistant's context again (it should only be in the User prompt), causing the model to attempt to continue text rather than generate audio tokens.
~~2. Non-deterministic Ordering: Using yaml.dump(..., sort_keys=True) randomizes the sequence of technical metadata (BPM, Key, Duration, etc.). Audio LLMs rely on a strict token order; this alphabetical shuffle disrupts the model's expectation compared to the fixed string formatting used in v0.12.2.~~ is incorrect. Apologies, I was trying to save time on the source code review.

I have attached two audio files for comparison:

[v0.12.2_good.mp3](https://github.com/user-attachments/files/25195217/v0.12.2_good.mp3): The expected quality.

[v0.12.3_glitchy.mp3](https://github.com/user-attachments/files/25195220/v0.12.3_glitchy.mp3): The result with artifacts.

### Logs

```powershell
The debug logs are too long to be pasted directly here.
I have attached the log file in the "Other" section
```

### Other

[comfyui_8000.log](https://github.com/user-attachments/files/25195386/comfyui_8000.log)

Working Version: ComfyUI Portable v0.12.2
Broken Version: ComfyUI Desktop v0.8.3 (Core ComfyUI v0.12.3)
OS: Windows 11
GPU: Nvidia 5060 Ti (16GB VRAM)
RAM: 80 GB

**Comment by djdookie** [comment]
I second this. I also have many artifacts in ACE-Step 1.5 generated songs now. This didn't happen some days ago.

**Comment by blepping** [comment]
> Redundant Context Injection: The user_metas dictionary incorrectly includes "caption".

That was my fault. I submitted a pull that added it (along with actually fixing some other stuff, in my defense). That has been corrected in the latest git revision.

> Non-deterministic Ordering: Using yaml.dump(..., sort_keys=True) randomizes the sequence of technical metadata (BPM, Key, Duration, etc.). Audio LLMs rely on a strict token order; this alphabetical shuffle disrupts the model's expectation compared to the fixed string formatting used in v0.12.2.

Sorting keys alphabetically is deterministic. This also matches the behavior of the official implementation which uses the same approach. The model's "expectation" is based on what it was trained on, it doesn't know/care what ComfyUI was doing a few days ago. Presumably the model was trained with the approach the official implementation uses.

> (Analysis assisted by GLM 4.7)

GLM doesn't know what it's talking about here. Be skeptical of what LLMs say unless you actually know enough about the subject to evaluate it yourself. They will basically always say something that sounds plausible and they'll say it confidently, even if it makes no actual sense.

***

*quick edit*: Unfortunately, these changes to improve conformance with the official implementation _will_ change seeds/generations. Most AI generations (in my opinion) are bad, you have to search/work to find the good ones. If, for example, one out of 10 generations is good then you have a 90% chance of getting a worse generation when something changes the seed. If the change improves average performance by 30% (random number, just for example) then you _still_ are very likely to have your existing seeds produce worse results.

I feel the pain of having seeds change and I was aware that my pull would change seeds. I didn't do it lightly, but now is the best time to fix these issues before there are even more workflows that would get broken.

Depending on how much effort you want to put into it, I have a set of nodes for ACE 1.0 and 1.5 here: https://gist.github.com/blepping/d0f6a26b1f59ed705999945821a3ee8a

That has a `RawTextEncodeAce15` node that lets you _exactly_ control how every prompt used by Ace15 gets tokenized. You would need to revert to the version that produced the results you want, add debug code to print out the prompts before they get tokenized and then use the exact prompts in that node which is obviously a lot of effort but it is an option if you really need to reproduce a workflow.

To make ComfyUI dump the prompts you would have to edit this code: https://github.com/Comfy-Org/ComfyUI/blob/a4be04c5d750cc5d62256f7f86bb5a7c0a78e28d/comfy/text_encoders/ace15.py#L225

Basically:
```python
        }

        # Just below the assignment to llm_prompts.
        # Make sure you preserve indentation.
        print(f"\n* Ace15 prompt debug:\n")
        for pk, pv in llm_prompts.items():
            print(f"*** {pk}\n{pv}\n")

        out = {
```

**Disclaimer**: Don't make changes you don't understand and make sure you save backups/know how to revert the change if something goes wrong.

**Comment by scottmudge** [comment]
I posted another issue (https://github.com/Comfy-Org/ComfyUI/issues/12399), but using the official ComfyUI workflows, I found the Ace Step 1.5 DiT model has the '.training' attribute set (both `model.model` and `model.model.model_sampling`), which means it has not been set to `.eval()` mode, which I believe can potentially cause this degradation issue.

I created a simple node to force `.eval()` on the model after the Diffusion Model loader, and it seemed to fix the issue for me. Or at least improved it a bit.

```
class ForceEvalMode:
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "model": ("*", {}),
            }
        }
    
    RETURN_TYPES = ("*",)
    RETURN_NAMES = ("model",)
    FUNCTION = "force_eval"
    
    def force_eval(self, model):
        # Try to find the actual PyTorch module and set it to eval
        if isinstance(model, torch.nn.Module):
            model.eval()
            print("[ForceEvalMode] Set model to eval mode")
        elif hasattr(model, 'model'):
            if isinstance(model.model, torch.nn.Module):
                model.model.eval()
                print("[ForceEvalMode] Set model.model to eval mode")
            if hasattr(model.model, 'model_sampling') and isinstance(model.model.model_sampling, torch.nn.Module):
                    model.model.model_sampling.eval()
                    print("[ForceEvalMode] Set model.model.model_sampling to eval mode")
        
        return (model,)
```

---

Edit: maybe ignore the above. `model.model` isn't in eval mode, but `model.model.diffusion_model` is, so that takes precedence. The mystery continues.

**Comment by djdookie** [comment]
I was able to narrow down the audio glitches to an custom node.
I'll leave a reference here in case someone else has the same issues and is trying to find a solution.
https://github.com/ryanontheinside/ComfyUI_RyanOnTheInside/issues/83#issuecomment-3881693796

**Comment by askmyteapot** [comment]
> The mystery continues.

I've come across the same issue where after maybe a couple dozen or so gens, the output becomes glitchy. But i havent been able to replicate it after doing 12 consecutive gens... but i have noticed that some seeds are just terrible.

**Comment by DB1052** [comment]
I've been using the comfyui default ace step 1.5 workflow in comfyui desktop. Everything worked fine before the latest update. This is what I get with the default workflow now (no changes to caption, lyrics, or settings) 

[ComfyUI Ace Step Workflow Generation.mp3](https://github.com/user-attachments/files/25225431/ComfyUI.Ace.Step.Workflow.Generation.mp3)

Edit: Fixed it. I saw the reply about Ryanontheinside nodepack (which i had installed). I uninstalled through manager, but I still got the same wonky result. Turns out, even though Manager showed it as uninstalled, it was still in the custom_nodes folder. Deleted it from there, and now it's working properly. Leaving this here in case anyone runs into the issue.

**Comment by Big-Whoop** [comment]
I confirm that the problem exists and for me the previous version was really spot on. I had hardly any bad generations. Now it's unusably bad and not even close to the former quality.

I do not have Ryanontheinside installed, never had.

Here are a few of the generations before the update (workflow embedded), perhaps someone can confirm that it got way worse.

https://drive.google.com/drive/folders/1iJWqhZXEgSYOqYBuKMPQTRUZECQhy-JN?usp=drive_link

**Comment by MssdTheMark** [comment]
With the release of v0.13.0, the glitchy artifacts have disappeared. However, I haven't been able to generate any high-quality ambient tracks in 48 attempts. I am closing this issue as the reported bug (glitches) is fixed.

P.S. If anyone ends up here looking for references, here are a few actually successful ambient tracks generated on v0.12.2:

[40_00001(deep house ambient 105 6 a minor).mp3](https://github.com/user-attachments/files/25270931/40_00001.deep.house.ambient.105.6.a.minor.mp3)
[41_00001(deep house ambient 105 6 a minor)_.mp3](https://github.com/user-attachments/files/25270944/41_00001.deep.house.ambient.105.6.a.minor._.mp3)
[124_00002(Electronic Minimalism 70 3 B minor)_.mp3](https://github.com/user-attachments/files/25270932/124_00002.Electronic.Minimalism.70.3.B.minor._.mp3)

And one not bad synthwave track from v0.13.0:

[191(retro-futuristic synthwave 120 6 E minor)_ace15_AIO_00001_.mp3](https://github.com/user-attachments/files/25270945/191.retro-futuristic.synthwave.120.6.E.minor._ace15_AIO_00001_.mp3)

Upd. Uploaded these tracks in original .flac format to Google Drive.
https://drive.google.com/drive/folders/1fdhnnvukX5y_JtpEsIuS7M4W06OZSETW?usp=sharing
